bazel_tools/javadoc_library.bzl

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This is a mixture of the javadoc_library rule in
+# https://github.com/google/bazel-common and the one in
+# https://github.com/stackb/rules_proto.
+
+# The main two factors for why we need this are
+
+# 1. We only have a source jar for java_proto_library.
+# The rules in bazel-common do not support this.
+
+# 2. We need to get the source jar via the JavaInfo provider.
+# The rules in rules_proto do not support this.
+
+# We probably want to clean this up and upstream it at some point.
+
+def _javadoc_library(ctx):
+    transitive_deps = []
+    for dep in ctx.attr.deps:
+        if JavaInfo in dep:
+            transitive_deps.append(dep[JavaInfo].transitive_deps)
+        elif hasattr(dep, "java"):
+            transitive_deps.append(dep.java.transitive_deps)
+
+    classpath = depset([], transitive = transitive_deps).to_list()
+
+    java_home = str(ctx.attr._jdk[java_common.JavaRuntimeInfo].java_home)
+
+    javadoc_command = [
+        java_home + "/bin/javadoc",
```

Comment:
Does this need a `.exe` on Windows?

Reply:
Afaik we don’t build any javadoc on Windows atm and I’m not sure we need to tbh. Either way, this is the same as the upstream rule (in both repos) so I’d suggest to merge it without Windows support and then add Windows support if we decide we want to have it given that this is blocking releases.